src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -1374,6 +1355,24 @@ class DropIndexStatementSegment(BaseSegment):
     )
 
 
+@mysql_dialect.segment()
+class DropRoutineStatementSegment(BaseSegment):
+    """A `DROP` statement that address stored procedures, stored functions, and loadable functions.
+
+    https://dev.mysql.com/doc/refman/8.0/en/drop-procedure.html
+    https://dev.mysql.com/doc/refman/8.0/en/drop-function-loadable.html
+    """
+
+    type = "drop_statement"
+    # DROP {PROCEDURE | FUNCTION} [IF EXISTS] {sp_name | function_name}
+    match_grammar = Sequence(
+        "DROP",
+        OneOf("PROCEDURE", "FUNCTION"),
+        Ref("IfExistsGrammar", optional=True),
+        Ref("ObjectReferenceSegment"),
+    )
```

Comment:
Could we split this out into separate function and procedure ones please. The function one will override ANSI.

Reply:
√